staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha4/api.proto

```diff
@@ -62,14 +62,31 @@ message NodePrepareResourcesResponse {
 
 message NodePrepareResourceResponse {
     // These are the additional devices that kubelet must
-    // make available via the container runtime. A resource
+    // make available via the container runtime. A claim
+    // may have zero or more requests and each request
     // may have zero or more devices.
-    repeated string cdi_devices = 1 [(gogoproto.customname) = "CDIDevices"];
+    repeated Device devices = 1;
     // If non-empty, preparing the ResourceClaim failed.
     // cdi_devices is ignored in that case.
     string error = 2;
 }
 
+message Device {
+    // The requests in the claim that this device is associated with.
+    // Optional. If empty, the device is associated with all requests.
+    repeated string request_names = 1;
+
+    // The pool which contains the device. Required.
+    string pool_name = 2;
+
+    // The device itself. Required.
+    string device_name = 3;
+
+    // A single device instance may map to several CDI device IDs.
+    // None is also valid.
+    repeated string cdi_device_ids = 4 [(gogoproto.customname) = "CDIDeviceIDs"];
+}
```

Comment:
This new `Device` struct is intentionally flexible. It provides enough information correlate CDI device IDs with requests (needed for containers) and with the device at the API level (needed for the pod resource API, see https://github.com/kubernetes/kubernetes/pull/125488#discussion_r1675607070).

Making `request_names` a list is intentional. We don't need this *right now* (once a device allocated for one request, it cannot be allocated again for another), but that is something that might change. For example, with "consumable resources" the same device might be used for two different requests.